CMakeLists.txt

```diff
@@ -194,7 +194,10 @@ set (IR_GENERATED_SRCS
   ${P4C_BINARY_DIR}/ir/ir-generated.h
   ${P4C_BINARY_DIR}/ir/ir-generated.cpp
   ${P4C_BINARY_DIR}/ir/gen-tree-macro.h)
-set (IR_GENERATOR ${P4C_BINARY_DIR}/tools/ir-generator/irgenerator)
+# IR_GENERATOR_DIRECTORY is used to set the RUNTIME_OUTPUT_DIRECTORY property
+# of the irgenerator target to the matching path
+set (IR_GENERATOR_DIRECTORY ${P4C_BINARY_DIR}/tools/ir-generator)
```

Comment:
This seems wrong -- the ir generator puts its generated files in  `${P4C_BINARY_DIR}/ir`.  The `${P4C_BINARY_DIR}/tools/ir-generator` is used for the lex&yacc output intermediate source files of the ir-generator.

Reply:
This path is where the `irgenerator` _executable_ is placed when it's built by a "non-IDE" build tool (make, or ninja).  When the build tool is `xcodebuild`, another subdirectory (Debug or Release) is normally appended to the path, depending on the build configuration.  Since the path is explicitly used by the custom command that runs `irgenerator` ( https://github.com/p4lang/p4c/blob/39deb73c51a1eaec8cbebcbb20f7aa2b69cc442f/CMakeLists.txt#L277-L283 ), it doesn't work when the cmake generator is Xcode (I assume it shouldn't work with Visual Studio either).  The change fixes the issue for Xcode cmake generator, but doesn't change anything for other generators (Unix Makefile or Ninja).  Whether it's wrong that `irgenerator` executable is placed in `${P4C_BINARY_DIR}/tools/ir-generator` or not -- I don't know -- this is how your `CMakeLists.txt` file has been for a while.

Reply:
So I guess its the comment that is confusing -- it implies that this is for the generated ir source files, not for the irgenerator binary.

Reply:
The `add_executable` cmake command uses the property `RUNTIME_OUTPUT_DIRECTORY` to determine the location of a built executable (a binary). Probably the name of the property is confusing. When I mention _output directory_ in this context I refer to the property `RUNTIME_OUTPUT_DIRECTORY` and the corresponding path where the executable (binary) is placed. This is _not_ where any output, produced by running the executable, may be placed.